applications/OptimizationApplication/python_scripts/output/container_variable_data_holder_output_decorator.py

```diff
@@ -0,0 +1,228 @@
+            elif isinstance(converted_data, KratosOA.ConditionPropertiesContainerVariableDataHolder):
+                converted_data = KratosOA.ConditionContainerVariableDataHolder(container_data)
+            elif isinstance(converted_data, KratosOA.ElementPropertiesContainerVariableDataHolder):
+                converted_data = KratosOA.ElementContainerVariableDataHolder(container_data)
+
+            # store the existing data to temp
+            temp = type(converted_data)(converted_data.GetModelPart())
+            temp.ReadDataFromContainerVariable(variable)
+
+            # assign data container
+            converted_data.AssignDataToContainerVariable(variable)
+
+            # store temp data in the dict
+            data["temp"] = temp
+
+    def ResetData(self):
+        for data in self.data_list:
+            data["temp"].AssignDataToContainerVariable(data["variable"])
+            del data["temp"]
+
+    @staticmethod
+    def GetContainerType(container: ContainerVariableDataHolderUnion) -> str:
+        if isinstance(container, KratosOA.NodalContainerVariableDataHolderBase):
+            return "nodes"
+        elif isinstance(container, KratosOA.ConditionContainerVariableDataHolderBase):
+            return "condition"
+        elif isinstance(container, KratosOA.ElementContainerVariableDataHolderBase):
+            return "elements"
+        else:
+            raise RuntimeError(f"The container type cannot be determined. [ Container = {str(container)}]")
+
+class ResponseContainerVariableDataHolderInfo(ContainerVariableDataHolderInfo):
+    @classmethod
+    def GetDefaultParameters(cls) -> Kratos.Parameters:
+        return Kratos.Parameters("""{
+            "type"           : "response_data",
+            "model_part_name": "PLEASE_PROVIDE_MODEL_PART_NAME",
+            "container_type" : "PLEASE_PROVIDE_CONTAINER_TYPE",
+            "response_name"  : "PLEASE_PROVIDE_RESPONSE_FUNCTION_NAME",
+            "data_list"      : [
+                {
+                    "data_name"     : "PLEASE_PROVIDE_DATA_NAME",
+                    "variable_name" : "PLEASE_PROVIDE_VARIABLE_NAME"
+                }
+            ]
+        }""")
+
+    @classmethod
+    def GetPattern(cls) -> str:
+        return "problem_data/response_data/<model_part_name>/<response_name>/sensitivities"
+
+class ControlContainerVariableDataHolderInfo(ContainerVariableDataHolderInfo):
+    @classmethod
+    def GetDefaultParameters(cls) -> Kratos.Parameters:
+        return Kratos.Parameters("""{
+            "type"           : "control_data",
+            "model_part_name": "PLEASE_PROVIDE_MODEL_PART_NAME",
+            "container_type" : "PLEASE_PROVIDE_CONTAINER_TYPE",
+            "control_name"   : "PLEASE_PROVIDE_CONTROL_NAME",
+            "data_list"      : [
+                {
+                    "data_name"     : "PLEASE_PROVIDE_DATA_NAME",
+                    "variable_name" : "PLEASE_PROVIDE_VARIABLE_NAME"
```

Comment:
this isn't so important, but I prefer empty defaults to invalid ones. If a default value is non-empty, it should be valid.

Also, giving an example of what kind of items you're expecting in `"data_list"` is neat, but it should be stated in the docstrings rather than the defaults. (Even better would be a system that generates valid defaults given a model, but that's out of scope for this PR).